app/components/preview_link_component/view.html.erb

```diff
@@ -0,0 +1,3 @@
+<% if @pages.any? %>
```

Comment:
theres a better way to do this sort of thing, let me dig up the docs for it

Reply:
Remove this condition in the markup and in the `view.rb` add the following method

```
def render?
   @pages.any
end
```

`render?` method should return a Boolean and based on that results ViewComponent will either render the component or not

docs: https://viewcomponent.org/api.html#render--boolean
example : https://www.honeybadger.io/blog/ruby-view-components/#conditional-rendering

Reply:
Very handy!